src/serialization/deserialize-assets.js

```diff
@@ -22,37 +22,29 @@ const deserializeSound = function (sound, runtime, zip, assetFileName) {
         return Promise.resolve(null);
     }
 
-    const assetId = sound.assetId;
-
-    // TODO Is there a faster way to check that this asset
-    // has already been initialized?
-    if (storage.get(assetId)) {
-        // This sound has already been cached.
-        return Promise.resolve(null);
-    }
     if (!zip) { // Zip will not be provided if loading project json from server
         return Promise.resolve(null);
     }
+
     const soundFile = zip.file(fileName);
     if (!soundFile) {
         log.error(`Could not find sound file associated with the ${sound.name} sound.`);
         return Promise.resolve(null);
     }
-    const dataFormat = sound.dataFormat.toLowerCase() === 'mp3' ?
-        storage.DataFormat.MP3 : storage.DataFormat.WAV;
+
     if (!JSZip.support.uint8array) {
         log.error('JSZip uint8array is not supported in this browser.');
         return Promise.resolve(null);
     }
 
-    return soundFile.async('uint8array').then(data => {
-        storage.builtinHelper.cache(
-            storage.AssetType.Sound,
-            dataFormat,
-            data,
-            assetId
-        );
-    });
+    const dataFormat = sound.dataFormat.toLowerCase() === 'mp3' ?
+        storage.DataFormat.MP3 : storage.DataFormat.WAV;
+    return soundFile.async('uint8array').then(data => storage.createAsset(
+        storage.AssetType.Sound,
+        dataFormat,
+        data,
+        sound.assetId
```

Comment:
A note for the future... should we be checking that the asset ids actually match? Maybe we can do that on the storage side when creating the asset...

Or is that something that happens on the server side...?

Reply:
The server will reject the asset upload if the contents don't match the md5 you send.